### PR TITLE
New variable for repo configuration, includes none as option

### PR DESCRIPTION
**Pull request by domenicbove**
# Description

Now supporting 3 repo options: confluent (default/current), custom, none (new)

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Validated the backwards compatibility logic by trying various variable options and debugging the variable

# Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: VARIABLES.md [hunk 1]
```diff
@@ -1178,6 +1146,30 @@ Below are the supported variables for the role confluent.common
...
+
+Configures package repositories on hosts. By default will configure confluent's deb/yum repositories. Possible options: none, confluent, custom. Must also set custom_yum_repofile_filepath or custom_apt_repo_filepath if using custom
+
+Default:  confluent
+
+***
+
+### custom_yum_repofile_filepath
+
+Full path on control node to custom yum repo file, must also set custom_yum_repofile to true
```
Comment (domenicbove) [inline VARIABLES.md:1159]: fix this description

File: VARIABLES.md [hunk 2]
```diff
@@ -1178,6 +1146,30 @@ Below are the supported variables for the role confluent.common
...
+
+Default:  ""
+
+***
+
+### custom_apt_repo_filepath
+
+Full path on control node to custom apt repo file, must also set custom_apt_repo to true
```
Comment (domenicbove) [inline VARIABLES.md:1167]: fix this description

File: roles/confluent.common/tasks/debian.yml [hunk 3]
```diff
@@ -111,7 +111,7 @@
   delay: 5
   when:
     - install_java|bool
-    - custom_apt_repo|bool
+    - repository_configuration == 'custom'
```
Comment (domenicbove) [inline roles/confluent.common/tasks/debian.yml:114]: think this should be, no matter what java needs to be installed...
```suggestion
    - repository_configuration in ['custom','none']
```

File: roles/confluent.common/defaults/main.yml [hunk 4]
```diff
@@ -1,4 +1,13 @@
 ---
+### Configures package repositories on hosts. By default will configure confluent's deb/yum repositories. Possible options: none, confluent, custom. Must also set custom_yum_repofile_filepath or custom_apt_repo_filepath if using custom
+repository_configuration: confluent
```
Comment (domenicbove) [inline roles/confluent.common/defaults/main.yml:3]: wonder if theres a way to make this var backwards compatible to the original vars

File: VARIABLES.md [hunk 5]
```diff
@@ -1178,6 +1146,30 @@ Below are the supported variables for the role confluent.common
 
 ***
 
+### repository_configuration
+
+Configures package repositories on hosts. By default will configure confluent's deb/yum repositories. Possible options: none, confluent, custom. Must also set custom_yum_repofile_filepath or custom_apt_repo_filepath if using custom. Note- vars custom_apt_repo and custom_yum_repofile are deprecated
```
Comment (JumaX) [inline VARIABLES.md:1151]: Minor grammar issue, change `Note-` to `Note:`

File: roles/confluent.common/defaults/main.yml [hunk 6]
```diff
@@ -1,4 +1,17 @@
 ---
+# Deprecated vars
+custom_apt_repo: false
+custom_yum_repofile: false
+
+### Configures package repositories on hosts. By default will configure confluent's deb/yum repositories. Possible options: none, confluent, custom. Must also set custom_yum_repofile_filepath or custom_apt_repo_filepath if using custom. Note- vars custom_apt_repo and custom_yum_repofile are deprecated
```
Comment (JumaX) [inline roles/confluent.common/defaults/main.yml:6]: Minor grammar issue, change Note- to Note:

**Comment by Fobhep** [comment]
LGTM - awesome :)

**Pull request merged**